tests/testthat/_snaps/verb-arrange.md

```diff
@@ -60,7 +60,7 @@
       lf %>% arrange(a) %>% select(-a) %>% mutate(c = lag(b))
     Output
       <SQL>
-      SELECT `b`, LAG(`b`, 1, NULL) OVER (ORDER BY `a`) AS `c`
+      SELECT `b`, LAG(`b`, 1, NULL) OVER () AS `c`
```

Comment:
This is a logical consequence of removing `a`. I think this makes more sense than the old behaviour.

Reply:
In the following code, I'm surprised we lose the `ORDER BY` in both cases. I think it makes sense to be able order a window function by a variable that doesn't appear in the final output. It's particularly surprising that we lose the `ORDER BY` in the window function, but not in the whole query.

``` r
library(dbplyr)
library(dplyr, warn.conflicts = FALSE)

lf <- lazy_frame(a = 1, b = 2, c = 3)

lf %>% window_order(a) %>% select(-a) %>% mutate(c = lag(b))
#> <SQL>
#> SELECT `b`, LAG(`b`, 1, NULL) OVER () AS `c`
#> FROM `df`
lf %>% arrange(a) %>% select(-a) %>% mutate(c = lag(b))
#> <SQL>
#> SELECT `b`, LAG(`b`, 1, NULL) OVER () AS `c`
#> FROM `df`
#> ORDER BY `a`
```

<sup>Created on 2023-01-26 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
The first query works by switching `select()` and `mutate()`

```r
lf %>% window_order(a) %>% mutate(c = lag(b)) %>% select(-a)
#> <SQL>
#> SELECT `b`, LAG(`b`, 1, NULL) OVER (ORDER BY `a`) AS `c`
#> FROM `df`
```

So, it is possible to order by a variable not in the final output.

How would you handle the following query?

```r
lf %>% window_order(a) %>% select(-a) %>% mutate(c = lag(b)) %>% mutate(c2 = lag(c))
```

This now produces:

```sql
<SQL>
SELECT *, LAG(`c`, 1, NULL) OVER () AS `c2`
FROM (
  SELECT `b`, LAG(`b`, 1, NULL) OVER () AS `c`
  FROM `df`
) `q01`
```

If we keep the window order for the inner `LAG()` we should keep it for the outer `LAG()`. This would imply adding `a` to the inner `SELECT` and then removing it in the outer `SELECT`. It should be doable but definitely complicates things.
Therefore, I would argue that you should use `mutate() %>% select(-a)` instead of `select(-a) %>% mutate()`.

Reply:
> It's particularly surprising that we lose the ORDER BY in the window function, but not in the whole query.

It is a bit surprising but it also makes sense because we "finish" the `arrange()` operation before removing `a`.